src/app/services/sync/voyager.service.ts

```diff
@@ -1,13 +1,37 @@
 import {
   createClient, VoyagerClient, DataSyncConfig,
-  OfflineQueueListener, ConflictListener, AuthContextProvider
+  OfflineQueueListener, ConflictListener, AuthContextProvider, NetworkStatus
 } from '@aerogear/voyager-client';
 import { Injectable, Injector } from '@angular/core';
 import { OpenShiftService } from '../openshift.service';
 import { AlertController } from '@ionic/angular';
 import { AuthService } from '../auth.service';
 import { taskCacheUpdates } from './cache.updates';
 
+export class ToggleableNetworkStatus implements NetworkStatus {
```

Comment:
To be moved to separate class.
The toggable state should be also persisted in ionic store.